src/bin/pg_autoctl/file_utils.c

```diff
@@ -161,7 +176,7 @@ bool
 write_file(char *data, long fileSize, const char *filePath)
 {
 
-	FILE *fileStream = fopen_with_umask(filePath, "wb", O_WRONLY | O_CREAT, 0644);
+	FILE *fileStream = fopen_with_umask(filePath, "wb", O_WRONLY | O_CREAT | O_EXCL, 0644);
```

Comment:
I think we should introduce a #define in defaults.h for FOPEN_FLAGS_W and FOPEN_FLAGS_A and then use the symbols rather than re-use always the same magic list. Also that gives a good place to comment on our O_EXCL usage here.

Reply:
O_EXCL actually doesn't make sense (which is why the tests failed). I thought it meant that only one process could open it, however it means that the file should not exist yet. Which we don't care about.